cmd/romana.k8s

```diff
@@ -0,0 +1,403 @@
+#!/bin/bash
+
+# Copyright (c) 2016 Pani Networks
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+# Usage subcommand.
+usage() {
+	echo "romana: command-line tool for Romana administration"
+	echo "Usage: romana <subcommmand> [options]"
+	echo "       romana <subcommmand> help"
+	echo "       romana [help]"
+	echo "Subcommands:"
+	echo "    add-host"
+	echo "    remove-host"
+	echo "    show-host"
+	echo "    create-owner"
+	echo "    delete-owner"
+	echo "    show-owner"
+	echo "    add-tier"
+	echo "    remove-tier"
+	echo "    add-policy"
+	echo "    remove-policy"
+	echo "    show-policy"
+}
+
+# Variables used within the script
+tenant_url=http://{{ stack_nodes.Controller.mgmt_ip }}:9602/
+tenant_path="tenants"
+topology_url=http://{{ stack_nodes.Controller.mgmt_ip }}:9603/
+topology_path="hosts"
+kube_apiserver_url=http://{{ stack_nodes.Controller.mgmt_ip }}:8080/
+romana_policy_path="apis/romana.io/demo/v1/namespaces/default/networkpolicys"
+
+add_host() {
+	if (( $# == 0 )) || [[ "$1" == "help" ]]; then
+		echo "romana add-host <hostname> <host-ip> <romana-cidr> <agent-port>"
+		return
+	fi
+	if (( $# != 4 )); then
+		echo "ERROR: expected 4 parameters, received $#"
+		return
+	fi
+	local hostname="$1" host_ip="$2" romana_cidr="$3" agent_port="$4"
+	# TODO: add validation for the parameters
+	post_data=$(printf '{"name": "%s", "ip": "%s", "romana_ip": "%s", "agent_port": %s}' "$hostname" "$host_ip" "$romana_cidr" "$agent_port")
+	echo "Sending request to '${topology_url}${topology_path}': '$post_data'"
+	curl -s -X POST -H "Content-Type: application/json" --data "$post_data" "${topology_url}${topology_path}"; printf "\n"
+}
+
+show_host() {
+	# Without arguments, show all the hosts
+	if (( $# == 0 )); then
+		show_hosts
+		return
+	fi
+	# Show usage.
+	# TODO: What if the host's name is 'help' ?
```

Comment:
At the moment it's effectively disallowing it, for simplicity.
I didn't want to burn much effort on this, adding option-parsing to every sub-command.